### ISSUE TITLE
Drop specs2 pre-v4 support

### ISSUE THREAD

**Issue by hmemcpy** [bazelbuild/rules_scala#767]
We're currently using a bit of a hack to support two API versions (using reflection) for specs2 v3 and v4.

Since rules_scala was upgraded to use specs2 v4, I propose dropping the older version support and removing reflection.

WDYT @ittaiz?

**Comment by johnynek** [comment]
It would be interesting to have stakeholders for each supported item. For instance, Stripe does not use specs. It would be good to have someone to reach out to when we are thinking of breaking changes.

AFAIK, Wix is the main stakeholder for specs + bazel.

**Comment by ittaiz** [comment]
@hmemcpy,
I agree. Especially given the last 3.x version was released a year ago
(June 18) and that the first 4.x was released ~2 years ago

@johnynek,
I think that would be super valuable. Any ideas how to maintain it?
markdown feels a bit off but maybe it's simplest.

**Comment by johnynek** [comment]
Maybe a more in depth markdown document would do it. We have adopters listed, but we don’t know how to contact them easily nor do we know what parts of the rules they use.

We could send emails to bazel-users and ask people to fill out a google sheets survey, we could set up a separate google group that would be very low traffic but we could use it to get feedback.

I don’t know what would be best.
